controllers/host/host_controller.go

```diff
@@ -1168,13 +1168,25 @@ func (r *HostReconciler) ReconcileHostByState(client *gophercloud.ServiceClient,
 }
 
 // hasAdminNetworkChange checks whether the addition of "admin" to PlatformNetworks
-// within an existing (multi-netting) EthernetInfo is present in the given profile and
-// current InterfaceInfo instances.
-func hasAdminNetworkChange(profileInterfaces *starlingxv1.InterfaceInfo, currentInterfaces *starlingxv1.InterfaceInfo) (*starlingxv1.EthernetInfo, bool) {
+// within an existing (multi-netting) EthernetInfo, VLANInfo and BondInfo is present
+// in the given profile and current InterfaceInfo instances.
+func hasAdminNetworkChange(profileInterfaces, currentInterfaces *starlingxv1.InterfaceInfo) (string, *starlingxv1.CommonInterfaceInfo, bool) {
 	if profileInterfaces == nil || currentInterfaces == nil {
-		return nil, false
+		return "", nil, false
 	}
 
+	if ethInfo, hasChange := hasEthernetAdminNetworkChange(profileInterfaces, currentInterfaces); hasChange {
+		return ethInfo.Name, &ethInfo.CommonInterfaceInfo, true
+	} else if VLANInfo, hasChange := hasVLANAdminNetworkChange(profileInterfaces, currentInterfaces); hasChange {
+		return VLANInfo.Name, &VLANInfo.CommonInterfaceInfo, true
+	} else if BondInfo, hasChange := hasBondAdminNetworkChange(profileInterfaces, currentInterfaces); hasChange {
+		return BondInfo.Name, &BondInfo.CommonInterfaceInfo, true
```

Comment:
I saw another IF type "VF" other than "eth" "vlan" "bond", do we need to also check that type?

Reply:
No, the mgmt or admin network cannot be of the VF type.

Reply:
acked